provider/cmd/cf2pulumi/schema-full.json

```diff
@@ -35477,7 +35477,8 @@
             },
             "type": "object"
         },
-        "aws-native:ecs:ClusterCapacityProviderAssociationsCapacityProvider": {
+        "aws-native:ecs:ClusterCapacityProviderAssociationsCapacityProvider0": {
```

Comment:
I don't see a `aws-native:ecs:ClusterCapacityProviderAssociationsCapacityProvider1`. If we don't have more then one variant, can we remove the trailing zero?

Reply:
Yeah, the other arm here is a primitive type so there isn't any `aws-native:ecs:ClusterCapacityProviderAssociationsCapacityProvider1` generated. Unfortunately, with the way the object types are generated, we have to add the digit before we know how many object types will be generated. I have one idea that might work for these simple cases though.

Reply:
I managed to avoid using the ugly type names by detecting this case and regenerating. We still generate the names though, so they clutter the schema a bit. I filed a TODO to track avoiding generating them to begin with. (I suspect it may require some pretty invasive changes to gen though.)